sdk/servicebus/Azure.Messaging.ServiceBus/src/Plugins/ServiceBusPlugin.cs

```diff
@@ -0,0 +1,204 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.Messaging.ServiceBus.Plugins
+{
+    /// <summary>
+    /// This class provides methods that can be overridden to manipulate messages for custom plugin functionality.
+    /// </summary>
+    public abstract class ServiceBusPlugin
+    {
+        /// <summary>
+        /// Gets the name of the <see cref="ServiceBusPlugin"/>.
+        /// </summary>
+        public abstract string Name { get; }
+
+        /// <summary>
+        /// Determines whether or an exception in the plugin should prevent a send or receive operation.
+        /// </summary>
+        public virtual bool ShouldContinueOnException => false;
```

Comment:
This feels like we're potentially hiding problems.  A lot of power is being granted to a property that may not convey the weight of the decision being made.

Should we consider making it the responsibility of the plugin author to ensure proper handling of exceptions and not surface them in a non-fatal scenario?

Reply:
Would also like to know what's the scenario for this.